scripts/patch-missing-contract-abis.ts

```diff
@@ -0,0 +1,121 @@
+import { connectPostgres, PgSqlClient } from '@hirosystems/api-toolkit';
+import { StacksCoreRpcClient } from '../src/core-rpc/client';
+import { getConnectionArgs, getConnectionConfig } from '../src/datastore/connection';
+import { ClarityAbi } from '../src/event-stream/contract-abi';
+import { loadDotEnv } from '../src/helpers';
+
+const BATCH_SIZE = 64;
+const LAST_BLOCK_HEIGHT = parseInt(process.env.LAST_BLOCK_HEIGHT ?? '-1');
+
+// 1) Environment + DB Setup
+loadDotEnv();
+const sql: PgSqlClient = await connectPostgres({
+  usageName: 'patch-missing-contract-abis',
+  connectionArgs: getConnectionArgs(),
+  connectionConfig: getConnectionConfig(),
+});
+
+try {
+  console.log('Starting script to patch missing contract ABIs...');
+
+  // 2) Initialize script variables and RPC client
+  let lastBlockHeight = LAST_BLOCK_HEIGHT; // Initial value for the first query
```

Comment:
Instead of pulling this from ENV you should be able to figure it out in your `missing` query (or with another query) to determine the block height of the first null contract.

Reply:
Happy to add, but I don't think that's needed since this only determines where the batches start from i.e. technically not needed if we don't batch., (e.g. useful for if the script does crash or stall or something, and we can pickup from the last block-height. But this also sort of assumes there may be "null" values that _should_ be there and are returned from the node, maybe that's also an incorrect assumption. Either way I think the default to -1 (or a similar value makes sense given some sorting).